test/float16.jl

```diff
@@ -74,12 +81,12 @@ end
 @testset "unary ops" begin
     @test -f === Float16(-2.)
     @test Float16(0.5f0)^2 ≈ Float16(0.5f0^2)
-    @test sin(f) ≈ sin(2f0)
-    @test log10(Float16(100)) == Float16(2.0)
-    @test sin(ComplexF16(f)) ≈ sin(complex(2f0))
-
-    # no domain error is thrown for negative values
-    @test cbrt(Float16(-1.0)) == -1.0
+    for func in (sin,cos,tan,asin,acos,atan,sinh,cosh,tanh,asinh,acosh,atanh,
+                 exp,exp2,exp10,expm1,log,log2,log10,log1p,sqrt,cbrt)
+        for x in FloatIterator{Float16}()
+            @test try func(x) === Float16(func(widen(x))) catch; true end
```

Comment:
```suggestion
            @test try; func(x); catch ex; ex; end === try; Float16(func(widen(x))); catch ex; ex; end
```
(style nit: don't write this many `;` on one line)

Reply:
That's not quite what I want. The errors are `DomainErrors` which I don't want to see. I only want to see the test failures.